runtime/nvqir/cutensornet/simulator_cutensornet.cpp

```diff
@@ -0,0 +1,376 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+#include "simulator_cutensornet.h"
+#include "cudaq.h"
+#include "cutensornet.h"
+#include "external_plugin.h"
+#include "tensornet_spin_op.h"
+
+namespace nvqir {
+static void initCuTensornetComm(cutensornetHandle_t cutnHandle) {
+#if defined CUDAQ_HAS_MPI
+  // Get the environment variable
+  const char *env_var = std::getenv("CUTENSORNET_COMM_LIB");
+  // If the variable is not set, use the one detected at compile time
```

Comment:
This environment variable is required by cutensornet as documented in its [docs](https://docs.nvidia.com/cuda/cuquantum/latest/cutensornet/api/functions.html#distributed-parallelization-api). 

After a bit of rethinking, I think the above logic of trying to set the variable via compile time detection could be error-prone, especially for deployment scenarios (not building from source).

I've revised this in https://github.com/NVIDIA/cuda-quantum/pull/863/commits/d67b1c0a8b5c577072d002cdf18e8f6bf01cdb7c :

- Better encapsulation of MPI supporting code into a separate object lib.

- When MPI is requested but `CUTENSORNET_COMM_LIB` is not set, log a warning, cutensornet distributed computation will likely be failing later. The steps to activate the distributed interface (via a script shipped with cuquantum) is well documented by cutensornet.

- Add an example.

Reply:
Much better! Just one quick thing that might help save someone from a lot of confused searching: Could you make sure CUTENSORNET_COMM_LIB is mentioned explicitly in the added documentation for this? Because the error a user will get mentions CUTENSORNET_COMM_LIB, so the natural thing to do is to search our docs for this. Hence we should make sure that when people do, they actually find the right section that gives them concrete steps to do.

Reply:
I've updated the documentation in PR#871: https://github.com/NVIDIA/cuda-quantum/pull/871/commits/4591d38a5bdc8d276ddb8d1a3648258c5c2b4206